hack/pre-release.sh

```diff
@@ -19,6 +23,30 @@ usage() {
   echo "  bumps the versions held back by CVP"
   echo "optional flags: -b"
   echo "  bumps the baseimage versions"
+  echo -e "${PINK}Requires connection to the RedHat VPN ${RESET}"
+}
+
+# This function increments any xyz version given an x y or z 
+# takes 2 arguments 
+# 1st arg: version to update in x.y.z format 
+# 2nd arg: stream to update. Can be x, y or z to represent the x y or z stream. 
+increment_version() {
+  if [ "$#" -ne 2 ]; then
+    echo incorrect parameter count for increment_version $#
+    return 1
+  fi
+  version=$1
+  # Setting the stream var to 1 2 or 3 which is used for incrementing the correct stream
+  # this tells awk which column to act on. Column 1, column 2, or column 3.
+  # which correspond to the x, y and z stream respectively.
+  declare -A stream_var
+  stream_var["x"]=1
+  stream_var["y"]=2
+  stream_var["z"]=3
+  stream_var=${stream_var[$2]} 
```

Comment:
Are you now overwriting the `stream_var` here? Won't using a new variable be more clear? Or is there some shell magic happening here which is not obvious?